### DISCUSSION TITLE
How to center text in a button

### DISCUSSION THREAD

**Discussion by zmarlon** [emilk/egui#5469] [Q&A]
Hello, I want to have multiple buttons with a fixed length. The text in this buttons should be centered. 

Im currently using this function to show a button:

```Rust
pub fn menu_button(ui: &mut Ui, text: &str) -> Response {
    egui::Button::new(egui::RichText::new(text).size(48.))
        .rounding(Rounding::ZERO)
        .min_size([250., 60.].into())
        .ui(ui)
}
```

Unfortunately, I have not yet found a way to ensure that the text in this fixed size button does not start on the left, but centered in the middle. How can this be realized?

**Comment by asinghka** [comment]
I have the same issue. Does anyone know how to center button text in egui?

**Comment by asinghka** [comment]
I figured it out. Here is how I centered the contents of a Button (this works with image and text contents):

```rust
ui.horizontal_centered(|ui| {
     egui::Button::new(egui::RichText::new(text).size(48.))
          .ui(ui)
}
```

**Comment by Codetheweb-true** [comment]
Working with egui often feels akin to dealing with black magic or voodoo, involving lots of guesswork and trial-and-error, If possible, it be advisable to abandon it entirely.

**Reply by mattfbacon** [reply]
To understand something if the docs aren't detailed enough, you can also read the code. It's usually pretty simple.
